poetry/semver/constraint_utils.py

```diff
@@ -0,0 +1,48 @@
+"""
+Constraint utilities
+
+The focus of this module is any constraint support, but
+PEP-0440 support trumps everything else.
+
+WRT, SEM-VER, PEP-0440 is partially but not fully
+compatible with all the SEM-VER specs.
+
+https://legacy.python.org/dev/peps/pep-0440/#semantic-versioning
+> Semantic versions containing a hyphen (pre-releases - clause 10) or
+> a plus sign (builds - clause 11) are not compatible with this PEP
+> and are not permitted in the public version field.
+"""
+
+from typing import List
```

Comment:
these imports aren't used, are they?

Reply:
My IDE is PyCharm-CE and helps to identify when to add these for typing annotations to work.  Using the `Code -> Optimize Imports` feature culls extra imports and these survive.